optaplanner-examples/src/main/java/org/optaplanner/examples/examination/solver/score/ExaminationConstraintProvider.java

```diff
@@ -0,0 +1,241 @@
+                .penalize("Two exams in a day",
+                        HardSoftScore.ONE_SOFT,
+                        (topicConflict, leftExam, config) -> topicConflict.getStudentSize() * config.getTwoInADayPenalty());
+    }
+
+    protected Constraint periodSpread(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(ExaminationConstraintConfiguration.class)
+                .join(TopicConflict.class)
+                .join(Exam.class,
+                        equal((config, topicConflict) -> topicConflict.getLeftTopic(), Exam::getTopic),
+                        filtering((config, topicConflict, leftExam) -> leftExam.getPeriod() != null))
+                .ifExists(Exam.class,
+                        equal((config, topicConflict, leftExam) -> topicConflict.getRightTopic(), Exam::getTopic),
+                        filtering((config, topicConflict, leftExam, rightExam) -> rightExam.getPeriod() != null),
+                        filtering((config, topicConflict, leftExam,
+                                rightExam) -> getPeriodIndexDifferenceBetweenExams(leftExam,
+                                        rightExam) < (config.getPeriodSpreadLength() + 1)))
+                .penalize("Period spread",
+                        HardSoftScore.ONE_SOFT,
+                        (config, topicConflict, leftExam) -> topicConflict.getStudentSize());
+    }
+
+    protected Constraint mixedDurations(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(Exam.class)
+                .ifExists(Exam.class, equal(Exam::getPeriod, Exam::getPeriod))
+                .ifNotExistsOther(Exam.class,
+                        equal(Exam::getPeriod, Exam::getPeriod),
+                        equal(Exam::getRoom, Exam::getRoom),
+                        lessThan(Exam::getId, Exam::getId))
+                .join(Exam.class,
+                        equal(Exam::getPeriod),
+                        equal(Exam::getRoom),
+                        greaterThan(Exam::getId),
+                        filtering((leftExam, rightExam) -> rightExam.getTopicDuration() != leftExam.getTopicDuration()))
+                .ifNotExists(Exam.class,
+                        equal((leftExam, rightExam) -> rightExam.getPeriod(), Exam::getPeriod),
+                        equal((leftExam, rightExam) -> rightExam.getRoom(), Exam::getRoom),
+                        equal((leftExam, rightExam) -> rightExam.getTopicDuration(), Exam::getTopicDuration),
+                        greaterThan((leftExam, rightExam) -> rightExam.getId(), Exam::getId))
+                .join(ExaminationConstraintConfiguration.class)
+                .penalize("Mixed durations", HardSoftScore.ONE_SOFT,
+                        (leftExam, rightExam, config) -> config.getMixedDurationPenalty());
+    }
+
+    protected Constraint frontLoad(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(ExaminationConstraintConfiguration.class)
```

Comment:
All uses of from/join on `ExaminationConstraintConfiguration` should theoretically be removable when converted to `penalizeConfigurable()`.